Rename CPUGenerator to CPUGeneratorImpl and CUDAGenerator to CUDAGeneratorImpl

pbelevich opened the pull request:
Fixes #34984

Stack from [ghstack](https://github.com/ezyang/ghstack):
* #36027 Remove operator-> from at::Generator
* **#36026 Rename CPUGenerator to CPUGeneratorImpl and CUDAGenerator to CUDAGeneratorImpl**

Differential Revision: [D20856458](https://our.internmc.facebook.com/intern/diff/D20856458)

pbelevich committed:
Rename CPUGenerator to CPUGeneratorImpl and CUDAGenerator to CUDAGeneratorImpl

[ghstack-poisoned]

pbelevich committed:
Update on "Rename CPUGenerator to CPUGeneratorImpl and CUDAGenerator to CUDAGeneratorImpl"




[ghstack-poisoned]

pbelevich committed:
Update on "Rename CPUGenerator to CPUGeneratorImpl and CUDAGenerator to CUDAGeneratorImpl"


Differential Revision: [D20856458](https://our.internmc.facebook.com/intern/diff/D20856458)

[ghstack-poisoned]

ezyang reviewed (approved):
Thanks. I know this sort of stuff is annoyingly grungy but I (and many readres of the codebase) really appreciate it!

pbelevich committed:
Update on "Rename CPUGenerator to CPUGeneratorImpl and CUDAGenerator to CUDAGeneratorImpl"


Differential Revision: [D20856458](https://our.internmc.facebook.com/intern/diff/D20856458)

[ghstack-poisoned]
